mcm/ValidationControl.py

```diff
@@ -146,39 +145,41 @@ def read_file_from_afs(self, path, trials_time_out=3):
 
         return out, err
 
-    def build_submission_command(self, prepid, run_test_path, timeout, memory):
-        timeout = int(timeout / 60.)
-        queue = self.QUEUE_1ND if (timeout / 3600. ) > 8. else self.QUEUE_8NH
-        memory = str(memory*1000) #we convert from MB to KB for batch
-        cmd = 'bsub -J ' + prepid
-        cmd += ' -g ' + self.group
-        cmd += ' -q ' + queue
-        cmd += ' -cwd ' + run_test_path
-        cmd += ' -W %s'%  timeout
-        run_test_path += '/' + self.TEST_FILE_NAME
-        cmd += ' -eo ' + run_test_path + '.err'
-        cmd += ' -oo ' + run_test_path + '.out'
-        cmd += ' -M ' + memory
-        cmd += ' bash ' + run_test_path
-        return cmd
-
     def check_ssh_outputs(self, stdin, stdout, stderr, fail_message):
         if not stdin and not stdout and not stderr:
             self.logger.error(fail_message)
             return False
         return True
 
-    def create_test_file(self, to_write, run_test_path):
+    def create_test_file(self, to_write, run_test_path, file_name):
         location = installer(run_test_path, care_on_existing=False, is_abs_path=True)
-        test_file_path = run_test_path + '/' + self.TEST_FILE_NAME
+        test_file_path = run_test_path + '/' + file_name
         try:
             with open(test_file_path, 'w') as there:
                 there.write(to_write)
             return True
         except Exception as e:
-            self.logger.error('There was a problem while creating the file: %s message: %s \ntraceback %s' % (run_test_path, str(e), traceback.format_exc()))
+            self.logger.error('There was a problem while creating the file: %s message: %s \ntraceback %s' % (test_file_path, str(e), traceback.format_exc()))
             return False
 
+    def create_htcondor_config_file(self, run_test_path, file_name, timeout, memory):
+        to_write = ''
+        validation_file = run_test_path + '/' + file_name
+        to_write += 'executable = %s\n' % validation_file
+        to_write += 'output = %s.out\n' % validation_file
+        to_write += 'error = %s.err\n' % validation_file
+        to_write += 'log = %s.log\n' % validation_file
+        to_write += '+MaxRuntime = %s\n' % timeout
+        to_write += 'RequestCpus = %s\n' % int(math.ceil(memory/2000.0)) # htcondor gives 2GB per core, if you want more memory you need to request more cores
```

Comment:
should be max(nThreads ) from sequence. although interesting how multicore cmDriver would run on single cpu